File: backend/app/Models/Publication.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Publication extends Model
{
    protected $table = 'publications';
    protected $primaryKey = 'publication_id';

    protected $fillable = [
        'title',
        'release_date',
        'pdf_file',
        'summary',
        'publication_category',
        'cover_url',
        'file_url',
        'description',
        'catalog_number',
        'publication_number',
        'isbn',
        'download_count',
        'is_featured',

    ];

    protected $casts = [
        'is_featured' => 'boolean'
    ];

    public $timestamps = true;

    public function category()
    {
        return $this->belongsTo(Category::class, 'publication_category');
    }

    public function getFileUrlAttribute($value)
    {
        return $value
            ? asset('storage/' . $value)
            : null;
    }
}

File: frontend/lib/models/publication.dart
class Publication {
  final int publicationId;
  final String title;
  final String? releaseDate;
  final String? summary;
  final String publicationCategory;
  final String? coverUrl;
  final String? description;
  final String? fileUrl;
  final String catalogNumber;
  final String publicationNumber;
  final String isbn;
  final int? downloadCount;

  Publication({
    required this.publicationId,
    required this.title,
    this.releaseDate,
    this.summary,
    required this.publicationCategory,
    this.coverUrl,
    this.description,
    this.fileUrl,
    required this.catalogNumber,
    required this.publicationNumber,
    required this.isbn,
    this.downloadCount,
  });

  factory Publication.fromJson(Map<String, dynamic> json) {
    return Publication(
      publicationId: int.parse(json['publication_id'].toString()),
      title: json['title']?.toString() ?? '',

      releaseDate: json['release_date']?.toString(),
      summary: json['summary']?.toString(),
      description: json['description']?.toString(),

      coverUrl: json['cover_url']?.toString(),
      fileUrl: json['file_url']?.toString(),

      publicationCategory: json['publication_category']?.toString() ?? '',

      catalogNumber: json['catalog_number']?.toString() ?? '',
      publicationNumber: json['publication_number']?.toString() ?? '',
      isbn: json['isbn']?.toString() ?? '',

      downloadCount:
          int.tryParse(json['download_count']?.toString() ?? ''),
    );
  }
}
